src/AudioPlayer.cpp

```diff
@@ -179,12 +179,8 @@ void AudioPlayer_SetInitVolume(uint8_t value) {
 
 void Audio_setTitle(const char *format, ...)
 {
-	// Allocates space for title of current track only once and keeps char* in order to avoid heap-fragmentation.
-	static char* _title = NULL;
-	if (_title == NULL) {
-		_title = (char *) x_malloc(sizeof(char) * 255);
-		gPlayProperties.title = _title;
-	}
+	static char _title[255];
+	gPlayProperties.title = _title;
```

Comment:
Keeping the title string on the heap is not really useful, so I would recommend to solve this by simply storing the whole char array in the global `gPlayProperties` struct (instead of a pointer).

Reply:
Yeah, that's the easiest and an elegant solution. I'll move the char array into the global structure and use it directly.

<sub>Damn, it so obvious, I should have seen it.</sub>